docs/se/guides/40_webclient.adoc

```diff
@@ -0,0 +1,420 @@
+
+[source,bash]
+.Run the greeting application first:
+----
+java -cp target/helidon-quickstart-se.jar io.helidon.examples.quickstart.se.Main
+----
+
+Open a new command prompt or terminal and run the ClientExample class you just created.
+
+[source,bash]
+.Run the greeting application first:
+----
+java -cp target/helidon-quickstart-se.jar io.helidon.examples.quickstart.se.ClientExample
+----
+
+[source,bash]
+.Output:
+----
+{"message":"Hello World!"}
+----
+
+When the ClientExample finishes its execution, you can stop the Main class by pressing `ctrl+c`.
+
+==== Discover other WebClient functionality
+
+In practice, String is not the most useful return type, since it usually needs some more handling.In this case it could
+be more interesting to return an object such as JsonObject. Earlier, Json support was added to the webclient,
+so it can be used instead of String.
+
+[source,java]
+.Replace String by JsonObject:
+----
+import javax.json.JsonObject;
+
+...
+
+webClient.get()
+    .path("/greet/David")
+    .request(JsonObject.class)  // <1>
+    .peek(System.out::println)
+    .await();
+----
+<1> Request a JsonObject as return value.
+
+In the Uri, the String following `greet` is a path parameter which allows the application to greet someone.
+
+[source,bash]
+.Output:
+----
+{"message":"Hello David!"}
+----
+
+This time, a JsonObject is printed out in the console. It is possible to change the greeting itself by using a PUT
+request to `/greet/greeting` endpoint from the base Uri.
+
+[source,java]
+.Modify the application greeting:
+----
+import javax.json.Json;
+
+...
+
+JsonObject entity = Json.createObjectBuilder() // <1>
+    .add("greeting", "Bonjour")
+    .build();
+webClient.put()     // <2>
+    .path("/greet/greeting")
+    .submit(entity)     // <3>
+    .thenCompose(response -> webClient.get()    // <4>
+        .path("/greet/David")
+        .request(JsonObject.class))
+    .thenAccept(System.out::println)
+    .await();
+----
+<1> Create a JsonObject with key `greeting` and value `bonjour`.
+<2> Create a PUT request.
+<3> Submit the JsonObject created earlier.
+<4> Once done, make a GET call to verify the modification was processed to the greeting.
+
+According to the quickstart documentation, a Json object can be sent to the application to change the greeting following
+this structure: `{"greeting" : "value"}`. The first three lines of code create the JsonObject with the required content.
+This time, we use the PUT request and submit methods to push the new greeting. One way to check the greeting modification
+is to execute GET request again and display obtained response. `thenCompose` method will execute a GET request after
+the PUT request is executed.
+
+[source,bash]
+.Output:
+----
+{"message":"Bonjour David!"}
+----
+
+=== WebClient Metrics
+
+WebClient as other Helidon components support Metrics. This chapter introduces the different metrics usable to measure
```

Comment:
WebClient, like other Helidon components, supports Metrics. The following example introduces the different metrics that can be used to measure WebClient activity. There are two ways to set up metrics: programmatically on the WebClient instance or manually using the configuration file.